### DISCUSSION TITLE
7-zip supports Mark Of The Web (MOTW) - How to enable it

### DISCUSSION THREAD

**Discussion by ghost** [beerisgood/Windows11_Hardening#11] [Q&A] [answered]
Source:
https://github.com/nmantani/archiver-MOTW-support-comparison#*1

> *1: Though 7-Zip has supported MOTW propagation since version 22.00, it is disabled by default. You can enable it for 7-Zip GUI with the "Propagate Zone Id stream:" option in "Tools" -> "Options" -> "7-Zip" of 7-Zip File Manager.
> 
> ![image](https://user-images.githubusercontent.com/72892766/191520687-cd30fd32-b3ba-45cf-b953-5c70bd565c37.png)
> 
> When you set the option to Yes, 7-Zip propagate MOTW to all extracted files. When you set it to "For Office files", 7-Zip propagate MOTW to files with the following file extensions:
> 
> .doc .docb .docm .docx .dot .dotm .dotx .wbk .wll .wwl
> .pot .potm .potx .ppa .ppam .pps .ppsm .ppsx .ppt .pptm .pptx .sldm .sldx
> .xla .xlam .xlm .xls .xlsb .xlsm .xlsx .xlt .xltm .xltx
> You can also enable MOTW propagation by setting the registry HKEY_CURRENT_USER\SOFTWARE\7-Zip\Options\WriteZoneIdExtract DWORD to 1.
> 
> For 7-Zip CLI, -snz switch is required to propagate MOTW regardless of the option above.

**Comment by beerisgood** [comment]
Windows internal ZIP doesn’t need any configuration and just works. 
7-Zip still is a joke for security.

**Reply by ghost** [reply]
I'd rather use Windows built in zip for everything too but sometimes I need to unpack files in .7zip or .rar formats and the built in extractor doesn't support them.

**Reply by beerisgood** [reply]
Fair Points 👍

**Reply by hemiipatu** [reply]
I hope this update for 7zip is a precursor to a more secure application in the future.

**Comment by SeasonalFerret** [comment]
Just fyi: This option does not show if you install using the .msi version of the program. 
However, it shows when using `winget install 7zip.7zip`, which uses the .exe version of the program.

I'm unsure if this means the .msi version is forced to support MOTW and has support enabled by default, or if this is just another reason to agree that 7-zip is a joke for security since now half of the downloads for windows doesn't even have this option.

**Reply by beerisgood** [reply]
So another point that 7Zip is just broken. 
thanks for your research! 🍺

**Comment by ghost** [comment, marked as answer]
This is the best alternative to 7-zip, **Nanazip**
https://github.com/M2Team/NanaZip

open-source, fork of 7-zip, digitally signed, beautiful UI, available in Microsoft Store, supports Mark of the web too.

![image](https://user-images.githubusercontent.com/72892766/201047802-b405cae4-ef31-4a99-876f-c9c32c436af2.png)

its right-click context menu also appears, by default, in the modern Windows 11 context menu too.

![image](https://user-images.githubusercontent.com/72892766/201048417-c32a0bc2-bd58-4e60-af42-97e117fb536e.png)


having something like Nanazip or 7-zip is essential for any pro user.

**Microsoft Store**
**Nanazip:** https://www.microsoft.com/store/productId/9N8G7TSCL18R
**Nanazip preview:** https://www.microsoft.com/store/productId/9NZL0LRP1BNL

**Reply by SeasonalFerret** [reply]
Dark mode planned for fall 2033

**Reply by hemiipatu** [reply]
I'd never heard of _NanaZip_ until you mentioned it, I'm glad you did, this is by far the best alternative to 7zip to-date!
